shared/middleware/assets_api.rb

```diff
@@ -77,21 +77,26 @@ def s3
   # Copy all files from one channel to another. Return metadata of copied files.
   #
   put %r{/v3/assets/([^/]+)$} do |encrypted_dest_channel_id|
-    src_owner_id, src_channel_id = storage_decrypt_channel_id(request.GET['src'])
-    dest_owner_id, dest_channel_id = storage_decrypt_channel_id(encrypted_dest_channel_id)
+    encrypted_src_channel_id = request.GET['src']
+    self.class.copy_assets(encrypted_src_channel_id, encrypted_dest_channel_id).to_json
+  end
+
+  def self.copy_assets(src_channel, dest_channel)
```

Comment:
Now that this code is being accessed from two places, this method should probably be extracted into a new class in shared/middleware/helpers/assets.rb . That should eliminate the need to change s3 and connect_s3 to class methods.